### PR TITLE
MINOR: replace FlattenedIterator by java stream

### PR DESCRIPTION
**Pull request by chia7712**
as title

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java [hunk 1]
```diff
@@ -183,8 +182,7 @@ public long brokerEpoch() {
 
     public Iterable<LeaderAndIsrPartitionState> partitionStates() {
         if (version() >= 2)
-            return () -> new FlattenedIterator<>(data.topicStates().iterator(),
-                topicState -> topicState.partitionStates().iterator());
+            return () -> data.topicStates().stream().flatMap(topicState -> topicState.partitionStates().stream()).iterator();
```
Comment (dajac) [inline clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java:185]: @chia7712 Is it also equivalent performance wise?

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java:185]: Indeed, we would want to verify that. It didn't look like it was when I originally checked, but I may have missed something.

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java:185]: We should be generally careful about replacing simple code paths with more complex ones (in terms of the underlying bytecode) in Kafka. Kafka is a high performance system, so code readability is one factor, but not the only one.

Comment (chia7712) [reply clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java:185]: @dajac @ijuma Thanks for your response. You are right. the performance should be more important than code readability. will close this PR :)

**Pull request closed without merging**